Modern/ops/src/main/java/org/bonej/ops/ellipsoid/FindEllipsoidFromBoundaryPoints.java

```diff
@@ -0,0 +1,343 @@
+package org.bonej.ops.ellipsoid;
+
+import net.imagej.ops.Op;
+import net.imagej.ops.special.function.AbstractBinaryFunctionOp;
+import net.imglib2.util.ValuePair;
+import org.joml.*;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+/**
+ * Tries to do one step of an {@link Ellipsoid} decomposition, starting from a point and a list of possible other points to collide with
+ * <p>
+ * The ellipsoid decomposition is drawn from the paper of Bischoff and Kobbelt, 2002 (https://www.graphics.rwth-aachen.de/media/papers/ellipsoids1.pdf)
+ * The variable naming widely follows Bischoff and Kobbelt's nomenclature, it is recommended to read this code in conjunction with the paper.
+ * The Ellipsoid-Plane intersection is done slightly differently, using the {@link EllipsoidPlaneIntersection}. This op starts growing an ellipsoid
+ * from the starting point p, and keeps growing in various directions until it hits three more points (q,r,s). If less than four distinct points are found,
+ * the op returns an empty ellipsoid. More details on the growing process are in the original publication.
+ * </p>
+ *
+ * @author Alessandro Felder
+ */
+@Plugin(type = Op.class)
+public class FindEllipsoidFromBoundaryPoints extends AbstractBinaryFunctionOp<List<ValuePair<Vector3d,Vector3d>>,Vector3d,Optional<Ellipsoid>> {
+
+    private static QuadricToEllipsoid quadricToEllipsoid = new QuadricToEllipsoid();
+    private static EllipsoidPlaneIntersection intersectionOp = new EllipsoidPlaneIntersection();
+
+    @Parameter(persist=false, required = false)
+    private double calibratedLargestImageDimension = 100.0;
+
+    @Override
+    public Optional<Ellipsoid>calculate(List<ValuePair<Vector3d,Vector3d>> fourVerticesWithNormals, Vector3d centre) {
+        List<Vector3d> fourVertices = new ArrayList<>();
+        fourVerticesWithNormals.forEach(v -> fourVertices.add(v.getA()));
+
+        fourVerticesWithNormals.sort(Comparator.comparingDouble(p->euclideanDistance(p.getA(),centre)));
+        VertexWithNormal p = new VertexWithNormal(fourVerticesWithNormals.get(0));
+        fourVertices.remove(p.getVertex());
+
+        ValuePair<VertexWithNormal, Double> qAndRadius = calculateQ(fourVertices, p);
```

Comment:
Since _Q1, Q2, Q3_ are quadrics, would it be simpler to express them with `Matrix4d`?